src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "ReportCu.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include "core/common/utils.h"
+
+namespace qr = xrt_core::query;
+
+enum class cu_stat : unsigned short {
+  usage = 0,
+  addr,
+  stat
+};
+
+uint32_t 
+parseComputeUnitStat(const std::vector<std::string>& custat, uint64_t offset, cu_stat kind) 
+{
+  uint32_t ret = 0;
+
+  for (auto& line : custat) {
+    uint32_t ba = 0, cnt = 0, sta = 0;
+#ifdef _WIN32
+    sscanf_s(line.c_str(), "CU[@0x%x] : %d status : %d", &ba, &cnt, &sta);
+#else
+    sscanf(line.c_str(), "CU[@0x%x] : %d status : %d", &ba, &cnt, &sta);
+#endif
+
+    if (offset != (uint64_t)ba)
+      continue;
+
+    if (kind == cu_stat::usage)
+      ret = cnt;
+    else if (kind == cu_stat::stat)
+      ret = sta;
+
+    return ret;
+  }
+
+  return ret;
+}
+
+boost::property_tree::ptree
+populate_cus(const xrt_core::device *device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  std::vector<char> ip_buf;
+  std::vector<std::string> cu_stats;
+
+  pt.put("description", desc);
+ 
+  try {
+    ip_buf = xrt_core::device_query<qr::ip_layout_raw>(device);
+    cu_stats = xrt_core::device_query<qr::kds_custats>(device);

```

Comment:
Why is the new query returning a vector of strings that need to be parsed in parseComputUnitStats to get the actually data. 

This doesn't make sense.  

Could you explain why this is needed as opposed to some other way to encode the data.